Userland/Libraries/LibWeb/Page/EditEventHandler.cpp

```diff
@@ -111,8 +111,15 @@ void EditEventHandler::handle_insert(JS::NonnullGCPtr<DOM::Position> position, u
         builder.append(node.data().bytes_as_string_view().substring_view(0, position->offset()));
         builder.append_code_point(code_point);
         builder.append(node.data().bytes_as_string_view().substring_view(position->offset()));
-        node.set_data(MUST(builder.to_string()));
 
+        // Cut string by max length
+        auto new_string = MUST(builder.to_string());
+        auto new_string_view = new_string.bytes_as_string_view();
+        if (node.max_length() >= 0 && static_cast<long>(new_string_view.length()) > node.max_length()) {
+            node.set_data(MUST(String::from_utf8(new_string_view.substring_view(0, min(new_string_view.length(), node.max_length())))));
```

Comment:
Hmm, looks pre-existing, but should these be byte offsets? Generally the spec either uses codepoint or codeunit offsets

Reply:
The code around it uses raw byte slices so I did the same, I'll at a comment that it needs to be converted to Unicode codepoints